crates/matrix-sdk/src/send_queue/upload.rs

```diff
@@ -349,6 +621,134 @@ impl QueueStorage {
         Ok(())
     }
 
+    /// Consumes a finished gallery upload and queues sending of the final
+    /// gallery event.
+    #[cfg(feature = "unstable-msc4274")]
+    #[allow(clippy::too_many_arguments)]
+    pub(super) async fn handle_dependent_finish_gallery_upload(
+        &self,
+        client: &Client,
+        event_txn: OwnedTransactionId,
+        parent_key: SentRequestKey,
+        mut local_echo: RoomMessageEventContent,
+        item_infos: Vec<FinishGalleryItemInfo>,
+        new_updates: &mut Vec<RoomSendQueueUpdate>,
+    ) -> Result<(), RoomSendQueueError> {
+        // All uploads are ready: enqueue the event with its final data.
+        let sent_gallery = parent_key
+            .into_media()
+            .ok_or(RoomSendQueueError::StorageError(RoomSendQueueStorageError::InvalidParentKey))?;
+
+        let mut sent_media_vec = sent_gallery.accumulated;
+        sent_media_vec.push(AccumulatedSentMediaInfo {
+            file: sent_gallery.file,
+            thumbnail: sent_gallery.thumbnail,
+        });
+
+        let mut sent_infos = HashMap::<String, AccumulatedSentMediaInfo>::new();
```

Comment:
1. Do we have to define the type of the hashmap here, or can Rust infer it?
2. Is it worth using a BTreeMap, for making the order of iteration deterministic?